// @next/nx-controls-common/framework.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { RouterModule } from '@angular/router';
import { FormsModule } from '@angular/forms';
import { MatSelectModule } from '@angular/material/select';
import { MatTableModule } from '@angular/material/table';
import { MatPaginatorModule } from '@angular/material/paginator';
import { MatSortModule } from '@angular/material/sort';
import { MatCheckboxModule } from '@angular/material/checkbox';
import { AlertComponent } from './components';
import { AlertCardComponent } from './components';
import { ConfirmDialogComponent } from './components';
import { FinderComponent } from './components';
import { MenuComponent } from './components';
import { SubMenuContentComponent } from './components';
import { KanbanComponent } from './components';
import { KanbanEmptyItemComponent } from './components';
import { AutocompleteComponent } from './components';
import { LoginStepsComponent } from './components';
import { SidenavComponent } from './components';
import { LineChartComponent } from './components';
import { DoughnutChartComponent } from './components';
import { SelectEditableComponent } from './components';
import { TitleEditableComponent } from './components';
import { NxLoginStepOneComponent } from './components/login-step-one/nx-login-step-one.component';
import { NxLoginStepTwoComponent } from './components/login-step-two/nx-login-step-two.component';
import { NxLoginStepThreeComponent } from './components/login-step-three/nx-login-step-three.component';
import { NxTableMasterDetailComponent } from './components/table-master-detail/table-master-detail.component';
import { NxTableRowDetailAComponent } from './components/table-row-detail-a/table-row-detail-a.component';
import { NxTableRowDetailBComponent } from './components/table-row-detail-b/table-row-detail-b.component';
import { AppService } from './services';
import { AlertService } from './services';
import { ConfirmationService } from './services';
import { NxLoginStepFourComponent } from './components/login-step-four/nx-login-step-four.component';
var CommonsModule = /** @class */ (function () {
    function CommonsModule() {
    }
    CommonsModule.forRoot = function () {
        return {
            ngModule: CommonsModule,
            providers: []
        };
    };
    CommonsModule.decorators = [
        { type: NgModule, args: [{
                    imports: [
                        CommonModule,
                        RouterModule,
                        FormsModule,
                        MatSelectModule,
                        MatTableModule,
                        MatPaginatorModule,
                        MatSortModule,
                        MatCheckboxModule
                    ],
                    providers: [AppService, AlertService, ConfirmationService],
                    declarations: [
                        AlertComponent,
                        AlertCardComponent,
                        ConfirmDialogComponent,
                        FinderComponent,
                        MenuComponent,
                        SubMenuContentComponent,
                        KanbanComponent,
                        KanbanEmptyItemComponent,
                        AutocompleteComponent,
                        LoginStepsComponent,
                        NxLoginStepOneComponent,
                        NxLoginStepTwoComponent,
                        NxLoginStepThreeComponent,
                        NxLoginStepFourComponent,
                        SidenavComponent,
                        LineChartComponent,
                        DoughnutChartComponent,
                        SelectEditableComponent,
                        TitleEditableComponent,
                        NxTableMasterDetailComponent,
                        NxTableRowDetailAComponent,
                        NxTableRowDetailBComponent
                    ],
                    exports: [
                        AlertComponent,
                        AlertCardComponent,
                        ConfirmDialogComponent,
                        FinderComponent,
                        MenuComponent,
                        SubMenuContentComponent,
                        KanbanComponent,
                        KanbanEmptyItemComponent,
                        AutocompleteComponent,
                        LoginStepsComponent,
                        NxLoginStepThreeComponent,
                        SidenavComponent,
                        LineChartComponent,
                        DoughnutChartComponent,
                        SelectEditableComponent,
                        TitleEditableComponent,
                        NxTableMasterDetailComponent,
                        NxTableRowDetailAComponent,
                        NxTableRowDetailBComponent
                    ]
                },] },
    ];
    /** @nocollapse */
    CommonsModule.ctorParameters = function () { return []; };
    return CommonsModule;
}());
export { CommonsModule };
