// @next/nx-controls-common/components/alert-card/alert-card.component.js
import { Component, Input } from '@angular/core';
import { AlertType } from '../../models/alert';
var AlertCardComponent = /** @class */ (function () {
    function AlertCardComponent() {
        this.isActive = false;
        this.isDead = false;
    }
    AlertCardComponent.prototype.ngOnInit = function () {
        var _this = this;
        setTimeout(function () {
            _this.isActive = true;
        }, 100);
        setTimeout(function () {
            _this.isActive = false;
        }, 3500);
        setTimeout(function () {
            _this.isDead = true;
        }, 4000);
    };
    AlertCardComponent.prototype.cssClass = function (alert) {
        if (!this.alert) {
            return;
        }
        // return css class based on alert type
        switch (this.alert.type) {
            case AlertType.Success:
                return 'success';
            case AlertType.Error:
                return 'error';
            case AlertType.Info:
                return 'info';
            case AlertType.Warning:
                return 'warning';
        }
    };
    // removeAlert(alert: Alert) {
    //   this.alerts = this.alerts.filter(x => x !== alert);
    // }
    AlertCardComponent.decorators = [
        { type: Component, args: [{
                    selector: 'nx-alert-card',
                    template: "\n\n          <div class=\"toast-wrapper\" [ngClass]=\"{ 'oss-toast-show': isActive == true, 'oss-toast-hide': isDead == true }\">\n            <div class=\"toast toast--{{ cssClass(alert) }} slideLeftRightFade-init slideLeftRightFade-show\">\n              <h3></h3>\n              <p>{{alert.message}}</p>\n          \n              <div class=\"toast-progressbar toast-progressbar--{{ cssClass(alert) }}\" style=\"width: 100%;\"></div>\n              <a></a>\n              <a></a>\n            </div>\n          </div>\n    "
                },] },
    ];
    /** @nocollapse */
    AlertCardComponent.ctorParameters = function () { return []; };
    AlertCardComponent.propDecorators = {
        'alert': [{ type: Input },],
    };
    return AlertCardComponent;
}());
export { AlertCardComponent };
