// @next/nx-core/ui-library/profile/profile.component.js
import { Component, Output, EventEmitter, Input } from '@angular/core';
import { Router, ActivatedRoute } from '@angular/router';
import { AuthService } from '../../services/auth.service';
import { FormControl } from '@angular/forms';
var ProfileComponent = /** @class */ (function () {
    function ProfileComponent(route, router, _authService) {
        this.route = route;
        this.router = router;
        this.logoutEvent = new EventEmitter();
        this.nxId = 'nx-profile-component';
        this.auth = false;
        this.username = new FormControl('');
        this.password = new FormControl('');
        this.isHide = true;
        this.showMenu = false;
        this.alias = '';
        if (_authService.IsAuth()) {
            this.auth = true;
        }
        this.authService = _authService;
    }
    ProfileComponent.prototype.ngOnInit = function () {
        this.errMsg = null;
        this.returnUrl = this.route.snapshot.queryParams['returnUrl'] || '/';
        var uSer = this.authService.getUser();
        if (uSer) {
            // this.alias = uSer.username.substring(0, uSer.username.lastIndexOf('@'));
            // const a = this.alias.split('.');
            // this.alias = a[0].substring(0, 1);
            this.alias = uSer.nombre.substring(0, 2);
            this.alias = this.alias.toUpperCase();
        }
    };
    ProfileComponent.prototype.logOut = function () {
        this.logoutEvent.emit('logout');
        // this.authService.logOut().subscribe(
        //   result => {
        //     localStorage.removeItem('currentUser');
        //     this.router.navigateByUrl('/login');
        //   },
        //   error => {
        //     switch (error.status) {
        //       case 401:
        //       case 403:
        //         this.errMsg = 'Verifique usuario y password e intente de nuevo.';
        //         break;
        //       case 0:
        //       case 404:
        //         this.errMsg =
        //           'Error en la comunicación con el servidor, favor de intentar mas tarde.';
        //         break;
        //       default:
        //         this.errMsg = 'Ocurrió un error, favor de intentarlo mas tarde';
        //         break;
        //     }
        //   }
        // );
    };
    ProfileComponent.prototype.toggleMenu = function () {
        this.showMenu = !this.showMenu;
    };
    ProfileComponent.decorators = [
        { type: Component, args: [{
                    selector: 'nx-profile',
                    template: "\n    <div id=\"{{nxId}}\" class=\"oss-profile\">\n      <!-- <span class=\"badge\"></span> -->\n      <h2>{{alias}}</h2>\n      <i class=\"icon ion-android-arrow-dropdown\" (click)=\"toggleMenu()\"></i>\n      <div class=\"oss-profile-menu\" *ngIf=\"showMenu\">\n        <ul>\n          <li (click)=\"logOut()\">\n            Cerrar sesi\u00F3n\n          </li>\n        </ul>\n      </div>\n    </div>\n  ",
                    styles: ["\n\n  "]
                },] },
    ];
    /** @nocollapse */
    ProfileComponent.ctorParameters = function () { return [
        { type: ActivatedRoute, },
        { type: Router, },
        { type: AuthService, },
    ]; };
    ProfileComponent.propDecorators = {
        'logoutEvent': [{ type: Output },],
        'nxId': [{ type: Input },],
    };
    return ProfileComponent;
}());
export { ProfileComponent };
