// @next/nx-core/ui-library/loading/loading.component.js
import { Component } from '@angular/core';
import { LoadingService } from '../../services/loading.service';
var LoadingComponent = /** @class */ (function () {
    function LoadingComponent(loadingService) {
        this.loadingService = loadingService;
    }
    LoadingComponent.prototype.ngOnInit = function () {
        this.isLoading = this.loadingService.isLoading;
    };
    LoadingComponent.decorators = [
        { type: Component, args: [{
                    selector: 'nx-loading',
                    template: "\n    <div class=\"modal\" [ngClass]=\"{'loading-modal-active': loadingService.isLoading}\">\n      <div class=\"oss-loading\">\n        <div class=\"oss-loading-content\">\n            <div class=\"oss-loading-circle\"></div>\n            <span class=\"oss-loading-name\"></span>\n        </div>\n      </div>\n    </div>\n  "
                },] },
    ];
    /** @nocollapse */
    LoadingComponent.ctorParameters = function () { return [
        { type: LoadingService, },
    ]; };
    return LoadingComponent;
}());
export { LoadingComponent };
