// @next/nx-controls-common/components/alert/alert.component.js
import { Component, Input } from '@angular/core';
import { AlertType } from '../../models/alert';
import { AlertService } from '../../services/alert.service';
var AlertComponent = /** @class */ (function () {
    function AlertComponent(alertService) {
        this.alertService = alertService;
        this.alerts = [];
        this.timeOut = 5;
        this.showing = false;
    }
    AlertComponent.prototype.ngOnInit = function () {
        var _this = this;
        this.alertService.getAlert().subscribe(function (alert) {
            if (!alert) {
                // clear alerts when an empty alert is received
                _this.alerts = [];
                return;
            }
            if (alert.type !== AlertType.Error) {
                // add alert to array
                _this.alerts.unshift(alert);
            }
            else {
                if (!_this.showing) {
                    _this.showing = true;
                    // add alert to array
                    _this.alerts.unshift(alert);
                    setTimeout(function () {
                        _this.showing = false;
                    }, _this.timeOut * 1000);
                }
            }
        });
    };
    AlertComponent.decorators = [
        { type: Component, args: [{
                    selector: 'nx-alert',
                    template: "\n    <div class=\"toast-container toast-container--slideLeftRightFade\" >\n      <nx-alert-card *ngFor=\"let alert of alerts\" [alert]=\"alert\"></nx-alert-card>\n    </div>\n  ",
                },] },
    ];
    /** @nocollapse */
    AlertComponent.ctorParameters = function () { return [
        { type: AlertService, },
    ]; };
    AlertComponent.propDecorators = {
        'timeOut': [{ type: Input },],
    };
    return AlertComponent;
}());
export { AlertComponent };
