// @next/nx-controls-common/components/doughnut-chart/doughnut-chart.component.js
import { Component, Input } from "@angular/core";
import { Chart } from "chart.js";
import { deepExtend } from "./helper";
var DoughnutChartComponent = /** @class */ (function () {
    function DoughnutChartComponent() {
        this.settings = {};
        this.nxId = 'nx-doughnut-chart';
        this.defaultSettings = {
            type: "pie",
            data: {
                labels: [],
                datasets: []
            },
            options: {
                legend: {
                    display: true
                },
                tooltips: {
                    backgroundColor: "#fff",
                    titleFontColor: "#000",
                    bodyFontColor: "#3c3c3c",
                    displayColors: false,
                    borderColor: "#d8d8d8",
                    borderWidth: 1
                }
            }
        };
    }
    DoughnutChartComponent.prototype.ngOnChanges = function (changes) {
        var params = this.prepareSettings();
        console.log(params);
        if (this.settings) {
            this.setChart(params);
        }
    };
    DoughnutChartComponent.prototype.setChart = function (settings) {
        this.chart = new Chart("doughnut-chart-canvas", settings);
    };
    DoughnutChartComponent.prototype.prepareSettings = function () {
        return deepExtend({}, this.defaultSettings, this.settings);
    };
    DoughnutChartComponent.decorators = [
        { type: Component, args: [{
                    selector: "nx-doughnut-chart",
                    template: "\n    <div *ngIf=\"chart\" id=\"{{nxId}}\">\n      <canvas id=\"doughnut-chart-canvas\">{{ chart }}</canvas>\n    </div>\n  "
                },] },
    ];
    /** @nocollapse */
    DoughnutChartComponent.ctorParameters = function () { return []; };
    DoughnutChartComponent.propDecorators = {
        'settings': [{ type: Input },],
        'nxId': [{ type: Input },],
    };
    return DoughnutChartComponent;
}());
export { DoughnutChartComponent };
