// @next/nx-core/auth/auth.guard.js
import { Injectable } from '@angular/core';
import { Router } from '@angular/router';
import { AuthService } from '../services/auth.service';
var AuthGuard = /** @class */ (function () {
    function AuthGuard(_router, _authService) {
        this.isLoggedIn = false;
        this.router = _router;
        this.authService = _authService;
    }
    AuthGuard.prototype.canActivate = function (route, state) {
        this.redirectURL = state.url;
        return this.checkIfCurrentUserIsAuthenticated();
    };
    AuthGuard.prototype.checkIfCurrentUserIsAuthenticated = function () {
        if (this.authService.IsAuth()) {
            return true;
        }
        else {
            this.router.navigate(['/login'], {
                queryParams: { returnUrl: this.redirectURL }
            });
            return false;
        }
    };
    AuthGuard.decorators = [
        { type: Injectable },
    ];
    /** @nocollapse */
    AuthGuard.ctorParameters = function () { return [
        { type: Router, },
        { type: AuthService, },
    ]; };
    return AuthGuard;
}());
export { AuthGuard };
