// @next/nx-controls-common/components/sidenav/sidenav.component.js
import { Component, Input, Output, EventEmitter } from '@angular/core';
var SidenavComponent = /** @class */ (function () {
    function SidenavComponent() {
        this.helpEvent = new EventEmitter();
        this.nxId = 'nx-sidenav';
        this.isSelected = false;
    }
    SidenavComponent.prototype.ngOnInit = function () { };
    SidenavComponent.prototype.toggleList = function (event) {
        var target = event.target || event.srcElement || event.currentTarget;
        var nestedList = target.parentElement;
        var listItems = nestedList.parentElement.childNodes;
        for (var _i = 0, listItems_1 = listItems; _i < listItems_1.length; _i++) {
            var element = listItems_1[_i];
            if (element.localName === 'li' && element.innerText !== nestedList.innerText) {
                element.classList.remove('selected');
                for (var _a = 0, _b = element.childNodes; _a < _b.length; _a++) {
                    var ul = _b[_a];
                    if (ul.localName === 'ul') {
                        ul.classList.remove('oss-list-active');
                    }
                }
            }
        }
        // toggle nested list
        if (nestedList.classList.contains('selected')) {
            nestedList.classList.remove('selected');
        }
        else {
            nestedList.classList.add('selected');
        }
        if (nestedList.children.length > 1) {
            var ul = nestedList.children[2];
            // console.log(ul);
            // toggle nested list
            if (ul.classList.contains('oss-list-active')) {
                ul.classList.remove('oss-list-active');
            }
            else {
                ul.classList.add('oss-list-active');
            }
        }
    };
    SidenavComponent.prototype.clickButton = function (e) {
        this.helpEvent.emit(e);
    };
    SidenavComponent.decorators = [
        { type: Component, args: [{
                    selector: 'nx-sidenav',
                    template: "\n    <div class=\"oss-extended-content\" id=\"{{nxId}}\">\n      <div class=\"oss-sidenav\">\n        <div class=\"oss-sidenav-logo\">\n          <img src=\"{{ logoUrl }}\">\n        </div>\n\n\n        <ul>\n          <li *ngFor=\"let i of collection; let idx = index;\" [ngClass]=\"{'selected': isSelected}\">\n            <ng-container *ngIf=\"i.items; then thenTemplateName; else elseTemplateName\">\n            </ng-container>\n\n            <ng-template #thenTemplateName>\n              <a id=\"nx-sidenav-item-{{ idx }}\" class=\"collapse\" (click)=\"toggleList($event)\">{{i.description}}</a>\n            </ng-template>\n\n            <ng-template #elseTemplateName>\n                <a id=\"nx-sidenav-item-{{ idx }}\" [routerLink]=[i.route] (click)=\"toggleList($event)\">\n                  {{i.description}}\n                </a>\n            </ng-template>\n\n            <ng-container *ngIf=\"i.items\">\n              <div class=\"arrowed\">\n                <span></span>\n              </div>\n              <ul class=\"oss-nested-list\">\n                <li *ngFor=\"let j of i.items; let jdx = index;\">\n                  <a id=\"nx-sidenav-subitem-{{ jdx }}\" [routerLink]=[j.route]>{{ j.description }}</a>\n                </li>\n              </ul>\n            </ng-container>\n          </li>\n        </ul>\n\n        <div class=\"oss-sidenav-button\">\n          <i class=\"icon ion-help-buoy\"></i>\n          <button (click)=\"clickButton($event)\">{{ buttonCaption }}</button>\n        </div>\n\n        <div class=\"oss-sidenav-banner\">\n          <img src=\"{{ logoBannerUrl }}\" alt=\"\">\n        </div>\n\n\n      </div>\n      <div class=\"oss-main-section\">\n        <ng-content></ng-content>\n      </div>\n    </div>\n  "
                },] },
    ];
    /** @nocollapse */
    SidenavComponent.ctorParameters = function () { return []; };
    SidenavComponent.propDecorators = {
        'collection': [{ type: Input },],
        'title': [{ type: Input },],
        'logoUrl': [{ type: Input },],
        'buttonCaption': [{ type: Input },],
        'logoBannerUrl': [{ type: Input },],
        'helpEvent': [{ type: Output },],
        'nxId': [{ type: Input },],
    };
    return SidenavComponent;
}());
export { SidenavComponent };
