// @next/nx-controls-common/components/login-step-three/nx-login-step-three.component.js
import { Component, Input, Output, EventEmitter, ViewChild } from '@angular/core';
var NxLoginStepThreeComponent = /** @class */ (function () {
    function NxLoginStepThreeComponent() {
        this.cancelEvent = new EventEmitter();
        this.nextEvent = new EventEmitter();
        this.geoFlag = false;
        this.lockMessage = {
            title: 'Se terminaron los intentos',
            paragraph: 'Por tu seguridad y disposiciones oficiales requerimos acceder a tu ubicación. Habilita esta opción desde tu navegador para acceder a tu Banca Electronica.'
        };
        this.inputSettings = {
            inputLabel: '',
            inputMinChars: 0,
            inputMaxChars: 0
        };
        this.errorMsg = '';
        this.errorMsgChange = new EventEmitter();
        this.logoUrl = '/assets/img/logo.png';
        this.contactInfo = [
            { place: 'Monterrey', phone: '2267 3446', description: '(BANREGIO)' },
            { place: 'Resto del país', phone: '01 81 2267 3446', description: '(BANREGIO)' },
            { place: 'EUA y Canadá', phone: '1 866 4320 550', description: '' }
        ];
        this.token = '';
        this.disabledBtn = true;
        this.unamePattern = '^[a-z0-9_-]{8,15}$';
    }
    NxLoginStepThreeComponent.prototype.ngOnInit = function () {
        this.trgFocusEl.nativeElement.focus();
    };
    NxLoginStepThreeComponent.prototype.cancelClick = function (e) {
        this.cancelEvent.emit(e);
    };
    NxLoginStepThreeComponent.prototype.nextClick = function (e) {
        this.nextEvent.emit(this.token);
    };
    NxLoginStepThreeComponent.prototype.keypressEvent = function (evento) {
        var e = evento;
        var validkeys = ['1', '2', '3', '4', '5', '6', '7', '8', '9', '0'];
        if (validkeys.indexOf(e.key) >= 0 || (e.keyCode === 13 || e.code === 'Enter')) {
            if ((e.keyCode === 13 || e.code === 'Enter') && this.disabledBtn === false) {
                this.nextClick(e);
            }
        }
        else {
            e.preventDefault();
        }
    };
    NxLoginStepThreeComponent.prototype.keyupEvent = function (e) {
        // const e = evento as KeyboardEvent;
        this.errorMsg = '';
        this.errorMsgChange.emit(this.errorMsg);
        if (this.token.length >= this.inputSettings.inputMinChars) {
            this.disabledBtn = false;
            if (e.keyCode === 13 || e.code === 'Enter') {
                this.nextClick(e);
            }
        }
        else {
            this.disabledBtn = true;
        }
        if (this.token.length > this.inputSettings.inputMaxChars) {
            this.token = this.token.substr(0, this.inputSettings.inputMaxChars);
        }
    };
    NxLoginStepThreeComponent.prototype.onPaste = function (e) {
        if (e && e.target) {
            this.token = e.target.value;
            e['keyCode'] = 0;
            this.keyupEvent(e);
        }
    };
    NxLoginStepThreeComponent.decorators = [
        { type: Component, args: [{
                    selector: 'nx-login-step-three',
                    template: "\n    <div class=\"login-container\">\n        <div class=\"login-banner\"><img src=\"{{ logoUrl }}\" alt=\"\" /></div>\n        <div class=\"login\">\n            <div  class=\"grid\">\n                <div class=\"col-12\">\n                    <div class=\"login-title\">\n                        <div class=\"login-steps\">\n                            <span> </span> <span> </span> <span class=\"active\"> </span>\n                        </div>\n                        <h2>Iniciar sesi\u00F3n</h2>\n                    </div>\n                </div>\n                <div class=\"col-12\">\n                    <div class=\"login-sub-title\">\n                        <h4>{{ inputSettings.inputLabel }}</h4>\n                    </div>\n                </div>\n                <div class=\"col-12\">\n                    <div class=\"login-content\">\n                        <input id=\"nx-login-token\" type=\"password\" [pattern]=\"unamePattern\" maxlength=\"{{ inputSettings.inputMaxChars }}\" [ngModel]=\"token\" (ngModelChange)=\"token = $event\" (keypress)=\"keypressEvent($event)\" (keyup)=\"keyupEvent($event)\" (paste)=\"onPaste($event)\"\n                            #userToken />\n                        <h5 class=\"login-error-message\" *ngIf=\"errorMsg\">{{ errorMsg }}</h5>\n                    </div>\n                </div>\n                <div class=\"col-12\">\n                    <div class=\"login-bottom\">\n                        <div class=\"nx-link dark\" (click)=\"cancelClick($event)\"> Cancelar Ingreso</div>\n                        <button id=\"nx-login-next-btn\" class=\"primary\" [disabled]=\"disabledBtn\" (click)=\"nextClick($event)\">Continuar</button>\n                        <button id=\"nx-login-next-btn\" class=\"default responsive no-soy-yo\" (click)=\"cancelClick($event)\">Cancelar Ingreso</button>\n                    </div>\n                </div>\n            </div>\n        </div>\n        <div class=\"login-footer\">\n            <div class=\"title\">\n                Si lo necesitas, ll\u00E1manos. Con gusto te atenderemos.\n            </div>\n            <div class=\"grid-3\">\n                <div class=\"col\" *ngFor=\"let item of contactInfo\">\n                    <span>{{ item.place }}</span>\n                    <span class=\"color-light\">{{ item.phone }}</span>\n                    <span>{{ item.description }}</span>\n                </div>\n            </div>\n        </div>\n    </div>\n  ",
                    styles: ["\n    .login-bottom{text-align:right}.login-bottom button{margin-left:10px}\n  "]
                },] },
    ];
    /** @nocollapse */
    NxLoginStepThreeComponent.ctorParameters = function () { return []; };
    NxLoginStepThreeComponent.propDecorators = {
        'trgFocusEl': [{ type: ViewChild, args: ['userToken',] },],
        'cancelEvent': [{ type: Output },],
        'nextEvent': [{ type: Output },],
        'geoFlag': [{ type: Input },],
        'userName': [{ type: Input },],
        'lockMessage': [{ type: Input },],
        'inputSettings': [{ type: Input },],
        'errorMsg': [{ type: Input },],
        'errorMsgChange': [{ type: Output },],
        'logoUrl': [{ type: Input },],
        'contactInfo': [{ type: Input },],
    };
    return NxLoginStepThreeComponent;
}());
export { NxLoginStepThreeComponent };
