// @next/nx-controls-common/components/sub-menu-content/sub-menu-content.component.js
import { Component, Input } from '@angular/core';
import { Router } from '@angular/router';
var SubMenuContentComponent = /** @class */ (function () {
    function SubMenuContentComponent(router) {
        this.router = router;
        this.nxId = 'nx-sub-menu-content';
        this.isSelected = false;
    }
    SubMenuContentComponent.prototype.ngOnInit = function () { };
    SubMenuContentComponent.prototype.toggleList = function (event) {
        var target = event.target || event.srcElement || event.currentTarget;
        var nestedList = target.parentElement;
        var listItems = nestedList.parentElement.childNodes;
        for (var _i = 0, listItems_1 = listItems; _i < listItems_1.length; _i++) {
            var element = listItems_1[_i];
            if (element.localName === 'li' && element.innerText !== nestedList.innerText) {
                element.classList.remove('selected');
                for (var _a = 0, _b = element.childNodes; _a < _b.length; _a++) {
                    var ul = _b[_a];
                    if (ul.localName === 'ul') {
                        ul.classList.remove('oss-list-active');
                    }
                }
            }
        }
        // toggle nested list
        if (nestedList.classList.contains('selected')) {
            nestedList.classList.remove('selected');
        }
        else {
            nestedList.classList.add('selected');
        }
        if (nestedList.children.length > 1) {
            var ul = nestedList.children[1];
            // console.log(ul);
            // toggle nested list
            if (ul.classList.contains('oss-list-active')) {
                ul.classList.remove('oss-list-active');
            }
            else {
                ul.classList.add('oss-list-active');
            }
        }
    };
    SubMenuContentComponent.decorators = [
        { type: Component, args: [{
                    selector: 'nx-sub-menu-content',
                    template: "\n    <div class=\"oss-extended-content\" id=\"{{nxId}}\">\n      <div class=\"oss-sub-menu\">\n        <h4>{{title.label}} <b>{{title.value}}</b></h4>\n        <div class=\"oss-progress-bar\">\n          <div class=\"oss-bar\">\n            <div class=\"oss-progress\" [ngStyle]=\"{'width.px': progress}\"></div>\n          </div>\n          <h6 class=\"oss-pb-label\">{{progress}}%</h6>\n        </div>\n\n        <ul>\n          <li *ngFor=\"let i of collection\" [ngClass]=\"{'selected': isSelected}\">\n            <ng-container *ngIf=\"i.items; then thenTemplateName; else elseTemplateName\">\n            </ng-container>\n\n            <ng-template #thenTemplateName>\n              <a class=\"collapse\" (click)=\"toggleList($event)\">{{i.description}}</a>\n            </ng-template>\n\n            <ng-template #elseTemplateName>\n                <a [routerLink]=[i.route] (click)=\"toggleList($event)\" [class.disabled]=\"i.disabled !== undefined && i.disabled === true\">\n                  {{i.description}}\n                </a>\n            </ng-template>\n\n            <ng-container *ngIf=\"i.items\">\n              <ul class=\"oss-nested-list\">\n                <li *ngFor=\"let j of i.items\">\n                  <a [routerLink]=[j.route] [class.disabled]=\"j.disabled !== undefined && j.disabled === true\">{{ j.description }}</a>\n                </li>\n              </ul>\n            </ng-container>\n          </li>\n        </ul>\n\n\n      </div>\n      <div class=\"oss-main-section\">\n        <ng-content></ng-content>\n      </div>\n    </div>\n  ",
                    styles: ["\n\n  "]
                },] },
    ];
    /** @nocollapse */
    SubMenuContentComponent.ctorParameters = function () { return [
        { type: Router, },
    ]; };
    SubMenuContentComponent.propDecorators = {
        'collection': [{ type: Input },],
        'title': [{ type: Input },],
        'progress': [{ type: Input },],
        'nxId': [{ type: Input },],
    };
    return SubMenuContentComponent;
}());
export { SubMenuContentComponent };
