// @next/nx-controls-common/components/line-chart/line-chart.component.js
import { Component, Input } from "@angular/core";
import { Chart } from "chart.js";
import { deepExtend } from "./helper";
var LineChartComponent = /** @class */ (function () {
    function LineChartComponent() {
        this.settings = {};
        this.nxId = 'nx-line-chart';
        this.defaultSettings = {
            type: "line",
            data: {
                labels: [],
                datasets: []
            },
            options: {
                elements: {
                    point: {
                        radius: 5,
                        backgroundColor: "#fff",
                        borderWidth: 2
                    },
                    line: {
                        borderWidth: 3
                    }
                },
                legend: {
                    display: false
                },
                scales: {
                    xAxes: [
                        {
                            display: true,
                            gridLines: {
                                display: false
                            }
                        }
                    ],
                    yAxes: [
                        {
                            display: true,
                            gridLines: {
                                borderDash: [4, 2]
                            }
                        }
                    ]
                },
                tooltips: {
                    backgroundColor: "#fff",
                    titleFontColor: "#000",
                    bodyFontColor: "#27CD6E",
                    displayColors: false,
                    borderColor: "#d8d8d8",
                    borderWidth: 1
                }
            }
        };
    }
    LineChartComponent.prototype.ngOnChanges = function (changes) {
        var params = this.prepareSettings();
        console.log(params);
        if (this.settings) {
            this.setChart(params);
        }
    };
    LineChartComponent.prototype.setChart = function (settings) {
        this.chart = new Chart("line-chart-canvas", settings);
    };
    LineChartComponent.prototype.prepareSettings = function () {
        return deepExtend({}, this.defaultSettings, this.settings);
    };
    LineChartComponent.decorators = [
        { type: Component, args: [{
                    selector: "nx-line-chart",
                    template: "\n    <div *ngIf=\"chart\" id=\"{{nxId}}\">\n      <canvas id=\"line-chart-canvas\">{{ chart }}</canvas>\n    </div>\n  "
                },] },
    ];
    /** @nocollapse */
    LineChartComponent.ctorParameters = function () { return []; };
    LineChartComponent.propDecorators = {
        'settings': [{ type: Input },],
        'nxId': [{ type: Input },],
    };
    return LineChartComponent;
}());
export { LineChartComponent };
