// @next/nx-core/framework.module.js
import { NgModule, CUSTOM_ELEMENTS_SCHEMA, ErrorHandler } from '@angular/core';
import { CommonModule } from '@angular/common';
import { HttpClientModule } from '@angular/common/http';
import { ReactiveFormsModule } from '@angular/forms';
import { AuthGuard } from './auth/auth.guard';
import { HTTP_INTERCEPTORS } from '@angular/common/http';
import { AuthInterceptor } from './auth/auth.interceptor';
import { AuthService } from './services/auth.service';
import { LoggerService } from './services/logger.service';
import { LoadingService } from './services/loading.service';
import { LoginService } from './services/login.service';
import { LoginComponent } from './ui-library/login/login.component';
import { ProfileComponent } from './ui-library/profile/profile.component';
import { ErrorManager } from './error/error-manager';
import { LoadingComponent } from './ui-library/loading/loading.component';
import { ErrorStreamService } from './services/error-stream.service';
import { NxInactivityDirective } from './directives';
import { CookieService } from 'ngx-cookie-service';
var FrameworkModule = /** @class */ (function () {
    function FrameworkModule() {
    }
    FrameworkModule.forRoot = function (config) {
        return {
            ngModule: FrameworkModule,
            providers: [
                AuthService,
                LoggerService,
                LoadingService,
                LoginService,
                ErrorStreamService,
                AuthGuard,
                CookieService,
                { provide: HTTP_INTERCEPTORS, useClass: AuthInterceptor, multi: true },
                { provide: ErrorHandler, useClass: ErrorManager },
                { provide: 'config', useValue: config }
            ]
        };
    };
    FrameworkModule.decorators = [
        { type: NgModule, args: [{
                    imports: [
                        CommonModule,
                        HttpClientModule,
                        // MatButtonModule,
                        // MatCheckboxModule,
                        // MatCardModule,
                        // MatFormFieldModule,
                        // MatSelectModule,
                        // MatInputModule,
                        // MatGridListModule,
                        // BrowserAnimationsModule,
                        ReactiveFormsModule
                    ],
                    exports: [
                        // KeyValueSearchableBoxComponent,
                        // NavBarComponent,
                        LoginComponent,
                        LoadingComponent,
                        ProfileComponent,
                        NxInactivityDirective
                    ],
                    declarations: [
                        // KeyValueSearchableBoxComponent,
                        // NavBarComponent,
                        LoginComponent,
                        LoadingComponent,
                        ProfileComponent,
                        NxInactivityDirective
                    ],
                    schemas: [CUSTOM_ELEMENTS_SCHEMA]
                },] },
    ];
    /** @nocollapse */
    FrameworkModule.ctorParameters = function () { return []; };
    return FrameworkModule;
}());
export { FrameworkModule };
