// @next/nx-core/services/auth.service.js
import { Injectable } from '@angular/core';
import { HttpHeaders, HttpClient } from '@angular/common/http';
import { CookieService } from 'ngx-cookie-service';
var AuthService = /** @class */ (function () {
    function AuthService(http, cookieService) {
        this.http = http;
        this.cookieService = cookieService;
        // var currentUser = JSON.parse(localStorage.getItem('currentUser'));
    }
    AuthService.prototype.IsAuth = function () {
        // if (localStorage.getItem('currentUser')) {
        if (!this.cookieService.check('currentUser')) {
            if (localStorage.getItem('currentUser')) {
                return true;
            }
            else {
                return false;
            }
        }
        else {
            return true;
        }
    };
    AuthService.prototype.getToken = function () {
        var currentUser = localStorage.getItem('currentUser');
        // const currentUser = this.cookieService.get('currentUser');
        if (currentUser) {
            return JSON.parse(currentUser).token.token;
        }
        else {
            return '';
        }
    };
    AuthService.prototype.getUser = function () {
        var currentUser = localStorage.getItem('currentUser');
        // const currentUser = this.cookieService.get('currentUser');
        if (currentUser) {
            return JSON.parse(currentUser);
        }
        else {
            return null;
        }
    };
    AuthService.prototype.getUserId = function () {
        var currentUser = localStorage.getItem('currentUser');
        // const currentUser = this.cookieService.get('currentUser');
        if (currentUser) {
            return JSON.parse(currentUser).userId;
        }
        else {
            return null;
        }
    };
    AuthService.prototype.Authenticate = function (userId, password, application) {
        var pass64 = btoa(password);
        var body = "{\"username\": \"" + userId + "\", \"password\": \"" + pass64 + "\", \"useCookie\": false, \"applicacion\": \"" + application + "\"}";
        var headers = new HttpHeaders().set('Content-Type', 'application/json');
        return this.http
            .post('/seguridad/login', body.toString(), {
            headers: headers,
            withCredentials: true
        });
    };
    AuthService.prototype.logOut = function () {
        var headers = new HttpHeaders().set('Content-Type', 'application/json');
        return this.http
            .delete('/seguridad/login', {
            headers: headers,
            withCredentials: true
        });
    };
    AuthService.decorators = [
        { type: Injectable },
    ];
    /** @nocollapse */
    AuthService.ctorParameters = function () { return [
        { type: HttpClient, },
        { type: CookieService, },
    ]; };
    return AuthService;
}());
export { AuthService };
