// @next/nx-controls-common/services/confirm.service.js
import { Injectable } from '@angular/core';
import { Subject } from 'rxjs/Subject';
var ConfirmationService = /** @class */ (function () {
    function ConfirmationService() {
        this.requireConfirmationSource = new Subject();
        this.acceptConfirmationSource = new Subject();
        this.requireConfirmation$ = this.requireConfirmationSource.asObservable();
        this.accept = this.acceptConfirmationSource.asObservable();
    }
    ConfirmationService.prototype.confirm = function (confirmation) {
        this.requireConfirmationSource.next(confirmation);
        return this;
    };
    ConfirmationService.prototype.close = function () {
        this.requireConfirmationSource.next({
            message: '',
            lblOkBtn: '',
            lblCancelBtn: '',
            close: true
        });
    };
    ConfirmationService.prototype.onAccept = function () {
        this.acceptConfirmationSource.next();
    };
    ConfirmationService.decorators = [
        { type: Injectable },
    ];
    /** @nocollapse */
    ConfirmationService.ctorParameters = function () { return []; };
    return ConfirmationService;
}());
export { ConfirmationService };
