// @next/nx-controls-common/components/table-row-detail-a/table-row-detail-a.component.js
import { Component, Input, Output, EventEmitter } from '@angular/core';
import { CurrencyPipe, DatePipe } from '@angular/common';
var NxTableRowDetailAComponent = /** @class */ (function () {
    function NxTableRowDetailAComponent() {
        this.clickEvent = new EventEmitter();
        this.detailKeys = [];
    }
    NxTableRowDetailAComponent.prototype.ngOnInit = function () {
        // console.log(this.content);
    };
    NxTableRowDetailAComponent.prototype.ngAfterViewInit = function () {
        var _this = this;
        if (this.content.detail) {
            this.config.detailColumns.forEach(function (key) {
                _this.detailKeys.push(key);
            });
        }
    };
    NxTableRowDetailAComponent.prototype.parseData = function (key) {
        var newVal = this.content.detail[key.column];
        switch (key.pipe) {
            case 'currency':
                newVal = newVal ? new CurrencyPipe('en-US').transform(newVal) : '';
                break;
            case 'date':
                newVal = newVal ? new DatePipe('en-US').transform(newVal, key.format) : '';
                break;
            default:
                newVal = newVal ? newVal.toString() : '';
                break;
        }
        return newVal.toString();
    };
    NxTableRowDetailAComponent.prototype.showClick = function (_event) {
        this.clickEvent.emit(_event);
    };
    NxTableRowDetailAComponent.decorators = [
        { type: Component, args: [{
                    selector: 'nx-table-row-detail-a',
                    template: "\n    <div class=\"grid\">\n      <div class=\"col\" *ngFor=\"let k of detailKeys\">{{ parseData(k) }}</div>\n      <div class=\"col\" (click)=\"showClick(content)\">\n        <i [class]=\"config.rowIconClass\"></i>\n      </div>\n    </div>\n  "
                },] },
    ];
    /** @nocollapse */
    NxTableRowDetailAComponent.ctorParameters = function () { return []; };
    NxTableRowDetailAComponent.propDecorators = {
        'content': [{ type: Input },],
        'config': [{ type: Input },],
        'clickEvent': [{ type: Output },],
    };
    return NxTableRowDetailAComponent;
}());
export { NxTableRowDetailAComponent };
